include/mav/MessageSet.h

```diff
@@ -70,6 +83,53 @@ namespace mav {
                 _document(std::move(document)),
                 _root_xml_folder_path(root_xml_folder_path) {}
 
+        
+        static inline uint64_t _strict_stoul(const std::string &str, int base=10) {
+            if (str.empty()) {
+                throw ParseError("Enum value is empty string");
+            }
+
+            size_t pos;
+            try {
+                uint64_t res = std::stoul(str, &pos, base);
+                if (pos != str.size()) {
+                    throw ParseError("Could not parse " + str + " as a number");
+                }
+                return res;
+            } catch (std::exception &e) {
+                throw ParseError("Could not parse " + str + " as a number (stoul failed): " + e.what());
+            }
+        }
+
+        static uint64_t _parseEnumValue(const std::string &str) {
+            // Check for binary format: 0b or 0B
+            if (str.size() >= 2 && (str.substr(0, 2) == "0b" || str.substr(0, 2) == "0B")) {
+                return std::bitset<64>(str.substr(2)).to_ullong();
+            }
+
+            // Check for hexadecimal format: 0x or 0X
+            if (str.size() >= 2 && (str.substr(0, 2) == "0x" || str.substr(0, 2) == "0X")) {
+                return _strict_stoul(str, 16);
```

Comment:
you could get rid of the first 2 characters?

Reply:
Done